assets/src/hooks/useRouteFilter.tsx

```diff
@@ -0,0 +1,93 @@
+import React from "react"
+import { useState } from "react"
+import { circleXIcon } from "../helpers/icon"
+import { Route } from "../skate"
+
+type FilterType = "id"
+
+export interface RouteFilter {
+  filterType: FilterType
+  filterText: string
+  handleTypeChange: (event: React.FormEvent<HTMLSelectElement>) => void
+  handleTextInput: (event: React.FormEvent<HTMLInputElement>) => void
+  clearTextInput: () => void
+}
+
+const isFilterType = (str: string): str is FilterType => str === "id"
+
+export const filterRoutes = (
+  allRoutes: Route[],
+  { filterType, filterText }: { filterType: FilterType; filterText: string }
+): Route[] => {
+  switch (filterType) {
+    case "id":
+      return allRoutes.filter((route: Route) => route.id.includes(filterText))
+    default:
+      return allRoutes
+  }
+}
+
+export const useRouteFilter = (): RouteFilter => {
+  const initialFilterType: FilterType = "id"
+  const [filterType, setFilterType] = useState(initialFilterType)
+  const [filterText, setFilterText] = useState("")
+
+  const handleTypeChange = (
+    event: React.FormEvent<HTMLSelectElement>
+  ): void => {
+    if (isFilterType(event.currentTarget.value)) {
+      setFilterType(event.currentTarget.value)
+    }
+  }
+
+  const handleTextInput = (event: React.FormEvent<HTMLInputElement>): void =>
+    setFilterText(event.currentTarget.value)
+
+  const clearTextInput = (): void => setFilterText("")
+
+  return {
+    filterType,
+    filterText,
+    handleTypeChange,
+    handleTextInput,
+    clearTextInput,
+  }
+}
+
+export const RouteFilterInterface = ({
```

Comment:
naming:

The component is named `RouteFilterInterface` and the object returned by the hook is named `RouteFilter`, which has confused me a couple of times. I think it might be clearer if the component was named simply `RouteFilter` (which makes it more like other components. We don't name the Route Picker `RoutePickerInterface`). The object could then be `Props` maybe? It's used as props but also has other uses. Maybe `RouteFilterData`?

Reply:
RouteFilterData makes sense to me.